src/pools/D3MAaveDaiPool.sol

```diff
@@ -0,0 +1,161 @@
+        variableDebt = ShareTokenLike(variableDebt_);
+        interestStrategy = interestStrategy_;
+        rewardsClaimer = RewardsClaimerLike(_rewardsClaimer);
+
+        ShareTokenLike(adai_).approve(pool_, type(uint256).max);
+        TokenLike(dai_).approve(pool_, type(uint256).max);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    // --- Math ---
+    function _add(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require((z = x + y) >= x, "DssDirectDepositHub/overflow");
+    }
+    function _mul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require(y == 0 || (z = x * y) / y == x, "D3MAaveDaiPool/overflow");
+    }
+    function _rdiv(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = _mul(x, RAY) / y;
+    }
+    function _min(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = x <= y ? x : y;
+    }
+
+    // --- Admin ---
+    function file(bytes32 what, address data) external auth {
+        require(live == 1, "D3MTestPool/no-file-not-live");
+
+        if (what == "king") king = data;
+        else revert("D3MPoolBase/file-unrecognized-param");
+    }
+
+    function validTarget() external view override returns (bool) {
+        (,,,,,,,,,, address strategy,) = LendingPoolLike(pool).getReserveData(address(asset));
+        return strategy == interestStrategy;
+    }
+
+    // Deposits Dai to Aave in exchange for adai which gets sent to the msg.sender
+    // Aave: https://docs.aave.com/developers/v/2.0/the-core-protocol/lendingpool#deposit
+    function deposit(uint256 amt) external override auth {
+        uint256 prevBalance = shareBalance();
+
+        LendingPoolLike(pool).deposit(address(asset), amt, address(this), 0);
+
+        // Verify the correct amount of shares shows up
+        require(shareBalance() >= _add(prevBalance, amt), "D3MAaveDaiPool/incorrect-share-credit");
+    }
+
+    // Withdraws Dai from Aave in exchange for adai
+    // Aave: https://docs.aave.com/developers/v/2.0/the-core-protocol/lendingpool#withdraw
+    function withdraw(uint256 amt) external override auth {
+        LendingPoolLike(pool).withdraw(address(asset), amt, address(hub));
+    }
+
+    // --- Collect any rewards ---
+    function collect(address[] memory assets, uint256 amount) external returns (uint256 amt) {
+        require(king != address(0), "D3MAaveDaiPool/king-not-set");
+
+        amt = rewardsClaimer.claimRewards(assets, amount, king);
+        emit Collect(king, assets, amt);
+    }
+
+    function transferShares(address dst, uint256 amt) public override returns (bool) {
+        return ShareTokenLike(adai).transfer(dst, amt);
+    }
+
+    function transferAllShares(address dst) external override returns (bool) {
+        return transferShares(dst, ShareTokenLike(adai).balanceOf(address(this)));
+    }
+
+    function accrueIfNeeded() external override {}
+
+    // --- Balance in standard ERC-20 denominations
```

Comment:
I don't really understand the meaning of this comment.
Not sure it's needed, but if it is I'd say something about this representing the underlying token (dai).